test/integration/src/peggy2/test_sifnode_transfers.py

```diff
@@ -0,0 +1,155 @@
+    test_sif_account_after_receive = ctx.get_sifchain_balance(test_sif_account)
+
+    assert test_sif_account_after_receive["rowan"] == amount_to_lock + test_sif_account_before_receive["rowan"]
+
+def test_double_peggy_token(ctx):
+    # Create/retrieve a test ethereum account
+    test_eth_account = ctx.create_and_fund_eth_account(fund_amount=fund_amount_eth)
+
+    # create/retrieve a test sifchain account
+    test_sif_account = ctx.create_sifchain_addr(
+        fund_amounts=[[fund_amount_sif, "rowan"], [fund_amount_eth, ctx.ceth_symbol],
+                      [fund_amount_double_peggy, double_peggy_symbol]])
+
+    destination_contract_address = ctx.get_destination_contract_address(double_peggy_address)
+
+    if destination_contract_address == eth.NULL_ADDRESS:
+        # write the ibc token entry to a json file
+        write_registry_json(double_peggy_symbol, double_peggy_address, isIbc=False)
+
+        # register the ibc token
+        ctx.sifnode.peggy2_token_registry_register_all(double_peggy_json_file_path, [0.5, "rowan"], 1.5, admin_account,
+                                                       ctx.sifnode_chain_id)
+        init_test_eth_account_balance = 0
+    else:
+        init_test_eth_account_balance = ctx.get_erc20_token_balance(destination_contract_address, test_eth_account)
+
+    ctx.sifnode_client.send_from_sifchain_to_ethereum(test_sif_account, test_eth_account, fund_amount_double_peggy, double_peggy_symbol)
+    ctx.advance_blocks()
+
+    if destination_contract_address == eth.NULL_ADDRESS:
+        created_contract_address = ctx.wait_for_new_bridge_token_created(double_peggy_address)
+        double_peggy_token_sc = ctx.get_generic_erc20_sc(created_contract_address)
+        final_test_eth_account_balance = ctx.get_erc20_token_balance(created_contract_address, test_eth_account)
+    else:
+        double_peggy_token_sc = ctx.get_generic_erc20_sc(destination_contract_address)
+        ctx.wait_for_eth_balance_change(test_eth_account, init_test_eth_account_balance, token_addr=destination_contract_address)
+        final_test_eth_account_balance = ctx.get_erc20_token_balance(destination_contract_address, test_eth_account)
+
+    assert init_test_eth_account_balance + fund_amount_double_peggy == final_test_eth_account_balance
+
+    init_test_sif_account_balance = ctx.get_sifchain_balance(test_sif_account)
+
+    ctx.send_from_ethereum_to_sifchain(test_eth_account, test_sif_account, fund_amount_double_peggy, token_sc=double_peggy_token_sc, isLock=False)
+    ctx.wait_for_sif_balance_change(test_sif_account, init_test_sif_account_balance, [[fund_amount_double_peggy, double_peggy_symbol]])
+    final_test_sif_account_balance = ctx.get_sifchain_balance(test_sif_account)
+    assert sifchain.balance_delta(init_test_sif_account_balance, final_test_sif_account_balance)[double_peggy_symbol] == fund_amount_double_peggy
+
+def test_ibc_to_eth_and_back_to_sifnode_transfer_valid(ctx):
+    # deploy erc20 for ibc token
```

Comment:
Yes, the PR is implemented without #2586. Since PR 2586 not reviewed yet and I am not sure if that solution will be merged. I will definitely re-implement ibc token export.

Reply:
Please have this branch merge in your #2586 branch so that those tests can be written.